### PR TITLE
Fixed min and max for KolInputDate

### LINKED ISSUE #6849: `_min` und `_max` werden beim `InputDate` nicht korrekt gesetzt
### Beschreibung:
Wird bei der `InputDate`-Component ein JavaScript `Date`-Objekt an die `_min` und `_max` Properties übergeben (like `_value`), werden die Werte nicht korrekt gesetzt – entweder werden sie gar nicht übernommen oder falsch geparst. Dies führt dazu, dass die erwarteten Eingabebeschränkungen (min/max) im UI nicht greifen.

### Aufgaben:
1. **Untersuchung des Problems:**
   - Überprüfen, warum die `_min` und `_max` Properties beim `InputDate` nicht korrekt geparst oder übernommen werden.
   - Sicherstellen, dass sowohl JS-`Date`-Objekte als auch Strings in verschiedenen `type`-Variationen (z.B. `date`, `datetime-local`, `month`, etc.) unterstützt und korrekt behandelt werden.

2. **Beispiel in der Sample-App:**
   - Ein Beispiel in der Sample-App integrieren, das die Nutzung von `_min` und `_max` mit einem JavaScript-`Date`-Objekt und Strings demonstriert.
   - Das Beispiel sollte verschiedene `type`-Varianten (z.B. `date`, `datetime-local`) beinhalten, um die Funktionalität vollständig abzudecken.

3. **Tests schreiben:**
   - Einen Test implementieren, der das Verhalten der `_min` und `_max` Properties mit folgenden Szenarien abtestet:
     - JS-`Date`-Objekte.
     - String-Repräsentationen von Daten.
   - Die Tests sollten für alle unterstützten `_type`-Variationen durchgeführt werden (z.B. `date`, `datetime-local`, `month`).

### Akzeptanzkriterien:
- Das `InputDate`-Component setzt korrekt die min/max Grenzen basierend auf JS-`Date`-Objekten und Strings.
- Ein Beispiel in der Sample-App zeigt die Funktionalität von min/max für verschiedene `type`-Varianten.
- Unit-Tests decken alle Kombinationen von `_min`/`_max` Werten und `type`-Variationen ab.

### PR DESCRIPTION
**Pull request by AlexanderSchmidtCE**

### CODE REVIEW

File: packages/components/src/components/input-date/controller.ts [hunk 1]
```diff
@@ -75,11 +76,38 @@ export class InputDateController extends InputIconController implements InputDat
 						return formattedTimeWithSeconds;
 					}
 				case 'week':
-					throw new Error('Auto convert to week is not supported!');
+					return `${formattedYear}-W${this.getWeekNumberOfDate(value)}`;
 			}
 		}
 	}
 
+	static getWeekNumberOfDate(date: Date): string {
```
Comment (sdvg) [inline packages/components/src/components/input-date/controller.ts:84]: (Vorschlag) Hier würden sich ein paar Unit-Tests anbieten, um die Edge-Cases abzusichern und die Methode besser zu dokumentieren.

File: packages/samples/react/e2e/input-date.spec.ts
Comment (sdvg) [inline packages/samples/react/e2e/input-date.spec.ts:1]: Diese Tests wären besser in Components aufgehoben. E2E Tests in der Sample App schreiben wir eigentlich nicht, höchstens für Scenarien.

Die entsprechende Spec existiert auch schon und müsste nur erweitert werden: `packages/components/src/components/input-date/input-date.e2e.ts:`

Comment (sdvg) [reply packages/samples/react/e2e/input-date.spec.ts:1]: Die Tests finde ich erst mal gut, allerdings vermisse ich die Szenarios mit Date-Objekten?

**Review by sdvg — CHANGES_REQUESTED** [review]

File: packages/samples/react/e2e/input-date.spec.ts
Comment (AlexanderSchmidtCE) [reply packages/samples/react/e2e/input-date.spec.ts:1]: Habe E2E Tests für die Komponente hinzugefügt.
Bei den E2E Tests in Samples navigiert playwright auf die entsprechende Seite und testet die Komponente mit der jeweiligen data-testid. Die Komponente mit den Date-Objekten wird in der MinMax.tsx erstellt.

File: packages/components/src/components/input-date/controller.ts (hunk 1 above)
Comment (AlexanderSchmidtCE) [reply packages/components/src/components/input-date/controller.ts:84]: Unit-Test ist erstellt

Comment (sdvg) [reply packages/components/src/components/input-date/controller.ts:84]: Hier hätte ich mir schon etwas mehr als einen Test vorgestellt :)

Wir haben doch recht komplexe Logik geschrieben, um die Kalenderwoche zu bestimmen, insbesondere mit den Edge-Cases zum Jahreswechsel. Gerade dazu hätte ich gerne Unit Tests gesehen, die diese Edge-Cases absichern und anhand von Beispielen dokumentieren.

File: packages/components/src/components/input-date/input-date.e2e.ts [hunk 2]
```diff
@@ -59,4 +59,142 @@ test.describe('kol-input-date', () => {
...
+			});
+			test('should set correct min and max for type date', async ({ page }) => {
+				const minDate = '2024-01-10';
+				const maxDate = '2024-10-20';
+				await page.setContent(`<kol-input-date _label="Date input" _type="date"></kol-input-date>`);
+
+				await page.locator('kol-input-date').evaluate((element: HTMLKolInputDateElement, date) => {
+					element._min = date;
+				}, minDateFormat);
+				await page.locator('kol-input-date').evaluate((element: HTMLKolInputDateElement, date) => {
```
Comment (sdvg) [inline packages/components/src/components/input-date/input-date.e2e.ts]: Ist nur eine Kleinigkeit, aber können wir min und max nicht jeweils in einer Evaluation setzen?

File: packages/components/src/components/input-date/controller.ts (hunk 1 above)
Comment (AlexanderSchmidtCE) [reply packages/components/src/components/input-date/controller.ts:84]: An die Edge-Cases zum Jahreswechsel hab ich nicht gedacht. Ich füge noch die hinzu

File: packages/components/src/components/input-date/input-date.e2e.ts (hunk 2 above)
Comment (AlexanderSchmidtCE) [reply packages/components/src/components/input-date/input-date.e2e.ts]: Guter Punkt

**Review by sdvg — APPROVED** [review]
